### PR TITLE
Add InView tracking (Intersection Observer/Scrolling to Elements)

### PR DESCRIPTION
**Pull request by darkruby501**
This PR introduces tracking for intersection events (component is visible in viewport) as part of the analytics infrastructure.

Tracking is added to the homepage and posts page.
Posts page tracking for reaching the comments section.
Home page tracking is of latestPosts section, recentDiscussion, and the load more buttons in each of those.
Also threw in tracking for a few more buttons in this PR, e.g. subscribeButtons.
I've got some concern about the number of events generated and want to think about how to handle that, in particular, the event tracker fires on and off when the page first loads while the post lists aren't yet populated. Probably I can clean that out of the data, but avoiding would also cut down on the number of events.

### CODE REVIEW

File: packages/lesswrong/lib/analyticsEvents.js [hunk 1]
```diff
@@ -32,7 +32,10 @@ export const AnalyticsUtil = {
 };
 
 export function captureEvent(eventType, eventProps) {
-  // console.log({eventType, eventProps}) //useful during development
+  if (eventType==="inViewEvent") {
+    const {time, inViewType, isIntersecting, intersectionRatio} = eventProps
+    console.log({eventType, time, inViewType, isIntersecting, intersectionRatio}) //useful during development
```
Comment (jimrandomh) [inline packages/lesswrong/lib/analyticsEvents.js]: Fails lint

File: packages/lesswrong/lib/analyticsEvents.js [hunk 2]
```diff
@@ -99,6 +102,27 @@ export function useTracking({eventType="unnamed", eventProps = {}, captureOnMoun
 
 export const withTracking = hookToHoc(useTracking)
 
+export function useIsInView({rootMargin='0px', threshold=0}={}) {
+  const [entry, setEntry] = useState(false)
+  const ref = useRef()
+  useEffect(() => {
+    console.log({rootMargin, threshold}) // save for debuggin
```
Comment (jimrandomh) [inline packages/lesswrong/lib/analyticsEvents.js]: Lint

Comment (darkruby501) [reply packages/lesswrong/lib/analyticsEvents.js]: The final PR, #2764 passes lint and has all console.log statements removed.

**Comment by jimrandomh** [comment]
This is a subset of https://github.com/LessWrong2/Lesswrong2/pull/2764 , closing to reduce confusion.

**Pull request closed without merging**